core/azure-core/src/main/java/com/azure/core/util/polling/Poller.java

```diff
@@ -0,0 +1,285 @@
+     * This is because poller does not know what to do in case on Mono.error.
+     * This function will return empty mono in case of Mono.error() returned by poll operation.
+     *
+     * @return mono of poll response
+     */
+    private Mono<PollResponse<T>> asyncPollRequestWithDelay() {
+        return  Mono.defer(() -> this.pollOperation.apply(this.pollResponse)
+            .delaySubscription(getCurrentDelay())
+            .onErrorResume(throwable -> {
+                // We should never get here and since we want to continue polling
+                //Log the error
+                return Mono.empty();
+            })
+            .doOnEach(pollResponseSignal -> {
+                if (pollResponseSignal.get() != null) {
+                    this.pollResponse = pollResponseSignal.get();
+                }
+            }));
+    }
+
+    private Duration getCurrentDelay() {
+        return (this.pollResponse != null && this.pollResponse.getRetryAfter() != null) ? this.pollResponse.getRetryAfter() : this.pollInterval;
+    }
+
+    /**
+     * Provide control to turn auto polling <strong>on or off</strong>. Once auto polling is turned off, it is <strong>user's responsibility</strong>
+     * to turn it back on.
+     *
+     * <p><strong>Disable auto polling</strong></p>
+     * {@codesnippet com.azure.core.util.polling.poller.disableautopolling}
+     *
+     * <p><strong>Enable auto polling</strong></p>
+     * {@codesnippet com.azure.core.util.polling.poller.enableautopolling}
+     *
+     * @param autoPollingEnabled true  Ensures the polling is happening in background.
+     * false Ensures that polling is <strong>not</strong> happening in background.
+     */
+    public final void setAutoPollingEnabled(boolean autoPollingEnabled) {
+
+        this.autoPollingEnabled = autoPollingEnabled;
+        if (this.autoPollingEnabled) {
+            if (!activeSubscriber()) {
+                this.fluxDisposable = this.fluxHandle.subscribe(pr -> this.pollResponse = pr);
+            }
+        } else {
+            if (activeSubscriber()) {
+                this.fluxDisposable.dispose();
+            }
+        }
+    }
+
+    /*
+     * Determine if this poller's internal subscriber exists and active.
+     */
+    private boolean activeSubscriber() {
+        return (this.fluxDisposable != null && !this.fluxDisposable.isDisposed());
+    }
+
+    /**
+     * Indicate if auto polling is <strong>on/off</strong> . By default auto polling is turned <strong>on</strong>.
+     *
+     * @return <strong>false</strong> if polling is stopped. <strong>true</strong> if auto polling is enabled.
+     */
+    public boolean isAutoPollingEnabled() {
+        return this.autoPollingEnabled;
+    }
+
+    /**
+     * Current known status as a result of last poll event. If auto polling was disabled by user, this will represent
+     * old response when auto polling was enabled.
```

Comment:
This is incorrect afaik - if auto-polling is disabled by user, this status will represent either the last auto-polled response, or the last response from a manual poll() call, whichever happened most recently.